src/style-spec/reference/v8.json

```diff
@@ -2959,7 +2959,7 @@
         }
       },
       "image": {
-        "doc": "Returns an `image` type for use in `icon-image`, `*-pattern` entries and as a section in the `format` expression. If set, the `image` argument will check that the requested image exists in the style and will return either the resolved image name or `null`, depending on whether or not the image is currently in the style. This validation process is synchronous and requires the image to have been added to the style before requesting it in the `image` argument.",
+        "doc": "Returns an `image` type for use in `icon-image`, `*-pattern` entries and as a section in the [`'format'`](#types-format) expression. A [`'coalesce'`](#coalesce) expression containing `image` expressions will evaluate to the first listed image that is currently in the style. This validation process is synchronous and requires the image to have been added to the style before requesting it in the `'image'` argument.",
```

Comment:
I think we should retain `If set, the image argument will check that the requested image exists in the style and will return either the resolved image name or null, depending on whether or not the image is currently in the style.` But we should update it to be more correct, e.g. `will either return the resolved image name [link to the ResolvedImage type https://docs.mapbox.com/mapbox-gl-js/style-spec/types/#resolvedimage] or will fire the styleimagemissing event, depending on whether or not the image is currently in the style.` That way we cover the bases of what happens for a stand-alone `image` expression and when used in `coalesce`.

Reply:
Made some changes here and in the [docs repo](https://github.com/mapbox/mapbox-gl-js-docs/pull/693), let me know what you think.